blocks/article-meta/article-meta.js

```diff
@@ -0,0 +1,177 @@
+import { getLibs } from '../../scripts/utils.js';
+
+async function validateAuthorUrl(url) {
+  if (!url) return null;
+
+  const resp = await fetch(`${url.toLowerCase()}.plain.html`);
+  if (!resp?.ok) {
+    console.log(`Could not retrieve metadata for ${url}`);
+    return null;
+  }
+
+  const html = await resp.text();
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(html, 'text/html');
+
+  return doc;
+}
+
+function openPopup(e) {
+  const target = e.target.closest('a');
+  const href = target.getAttribute('data-href');
+  const type = target.getAttribute('data-type');
+  window.open(
+    href,
+    type,
+    'popup,top=233,left=233,width=700,height=467',
+  );
+}
+
+async function buildAuthorInfo(authorEl, bylineContainer) {
+  const miloLibs = getLibs();
+  const { createTag, getConfig } = await import(`${miloLibs}/utils/utils.js`);
+
+  const { href, textContent } = authorEl;
+
+  const config = getConfig();
+  const base = config.miloLibs || config.codeRoot;
+  const authorImg = createTag('div', { class: 'article-author-image' });
+  authorImg.style.backgroundImage = `url(${base}/blocks/article-header/adobe-logo.svg)`;
+  bylineContainer.prepend(authorImg);
+
+  const doc = await validateAuthorUrl(href);
+  if (!doc) {
+    const p = createTag('p', null, textContent);
+    authorEl.replaceWith(p);
+    return;
+  }
+
+  const img = doc.querySelector('img');
+  if (img) {
+    img.setAttribute('alt', authorEl.textContent);
+    authorImg.append(img);
+    if (!img.complete) {
+      img.addEventListener('load', () => {
+        authorImg.style.backgroundImage = 'none';
+      });
+      img.addEventListener('error', () => {
+        img.remove();
+      });
+    } else {
+      authorImg.style.backgroundImage = 'none';
+    }
+  }
+}
+
+export async function initSharingLinkFunction(sharing) {
+  const miloLibs = getLibs();
+  const { getConfig } = await import(`${miloLibs}/utils/utils.js`);
+  const { replaceKey } = await import(`${miloLibs}/features/placeholders.js`);
+  const { copyToClipboard } = await import(`${miloLibs}/utils/tools.js`);
+
+  const sharingLinks = sharing.querySelectorAll('[data-href]');
+  if (!sharingLinks) return;
+
+  sharingLinks.forEach((link) => {
+    link.addEventListener('click', openPopup);
+  });
+  const copyButton = sharing.querySelector('#copy-to-clipboard');
+  copyButton.addEventListener('click', async () => {
+    const copyText = await replaceKey('copied-to-clipboard', getConfig());
+    await copyToClipboard(copyButton, copyText);
+  });
+}
+
+export async function buildSharing() {
```

Comment:
Thanks @rgclayton, do you know who can I contact with to grant me read access on Adobe Milo Sharepoint? 
It would be helpful to view the actual doc in sharepoint for https://milo.adobe.com/docs/authoring/examples/share-block to see how's implemented, thank you

Reply:
@snowiewdev, 

You will need an Adobe email to access SharePoint content. 
You should be able to view the document source from the sidekick plugin.

![image](https://github.com/user-attachments/assets/ae0cbc32-e425-4641-9baf-cf1ba81920f4)


You can also use the sidekick in your drafts folder and add the Share block to a page to see it in action.

![image](https://github.com/user-attachments/assets/170c1b4a-09f6-432e-980c-5403f318d14d)

Reply:
Thanks Ryan! The `view document source` from the plugin is super helpful. Updated `article-meta` to reuse the `share` block from milo. Let me know if it needs to updated further.